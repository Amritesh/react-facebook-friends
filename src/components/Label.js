import React, { Component, PropTypes } from 'react';
import Radium from 'radium';

const style = {
  label: {
    fontSize: '1.5em',
    textAlign: 'center',
    '@media (max-width: 48em)': {
      fontSize: '0.75em',
      display: 'none'
    }
  },
  value: {
    margin: '3px',
  },
  valueGroup: {
    fontSize: '4em',
    '@media (max-width: 48em)': {
      fontSize: '1em'
    }
  }
};

const Label = ({ text, icon, top, value }) => (
  <div style={{
    margin: value < 10 ? '0 25px' : '0 10px',
    display: 'inline-block',
    '@media (max-width: 48em)': {
      position: 'absolute',
      top: top,
      margin: value < 10 ? '0 25px' : '0 20px',
      backgroundColor: 'rgb(58, 87, 149)',
      color: 'white',
      padding: '5px',
      borderRadius: '3px'
    }
  }}>
    <div style={style.valueGroup}>
      <i className={icon}></i>
      <span style={style.value}>{value}</span>
    </div>
    <div style={style.label}>{text}</div>
  </div>
);

export default Radium(Label);
