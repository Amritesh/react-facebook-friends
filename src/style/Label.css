.label-wrapper {
  justify-content: center;
  flex-wrap: wrap;
  display: flex;
  padding: 0;
}

.label-text {
  font-size: 1.5em;
}

.label-valueGroup {
  font-size: 4em;
}

@media (max-width: 590px) {
  .label-wrapper {
    flex: 1 1 100%;
  }

  .label-text {
    display: none;
  }

  .label-valueGroup {
    font-size: 1em;
    color: white;
    background: rgb(58, 87, 149);
    padding: 3px;
    border-radius: 3px;
  }
}