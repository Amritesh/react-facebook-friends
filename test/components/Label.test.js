import React from 'react';
import renderer from 'react-test-renderer';
import Label from '../../src/components/Label';

test('Label component should renders correctly', () => {
  const props = {
    text: 'test',
    icon: 'icon',
    top: '5',
    value: '1'
  };
  const component = renderer.create(<Label {...props} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
