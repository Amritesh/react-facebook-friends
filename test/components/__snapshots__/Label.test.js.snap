// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component should renders correctly 1`] = `
<div
  className="label-wrapper"
>
  <div
    className="label-valueGroup"
  >
    <i
      className="icon"
    />
    <span>
      1
    </span>
  </div>
  <div
    className="label-text"
  >
    test
  </div>
</div>
`;
